plugin/pkg/replacer/replacer.go

```diff
@@ -22,8 +22,13 @@ type Replacer interface {
 	Set(key, value string)
 }
 
+// Func is the type of function in the metadata, when called they return the value of the label.
+type Func func() string
+
 type replacer struct {
-	replacements map[string]string
+	ctx          context.Context
+	replacements map[string]Func   // replacements are given as function for lazy init
```

Comment:
only the label replacements needs to be a function,  the rest can basically stay as-is?

Reply:
Yes. I was addressing the comment of JohnB in PR #2240 for lazy initialization. It avoid doing the computation for values that are not used in the log format.

Reply:
yes, but this only matters for labels, to we just needs a separate map just for that.

Reply:
ok. will come backward a little bit to extend only for labels.